src/puppetlabs/puppetdb/http/aggregate_event_counts.clj

```diff
@@ -9,25 +10,35 @@
 (defn routes
   [version]
   (app
-   [""]
-   {:get (fn [{:keys [params globals]}]
-           (let [{:strs [query summarize_by counts_filter count_by] :as query-params} params
-                 distinct-options (events-http/validate-distinct-options! query-params)
-                 query-options (-> {:counts_filter (when counts_filter (json/parse-string counts_filter true))
-                                    :count_by count_by}
-                                   (merge distinct-options))]
-             (produce-streaming-body
-              :aggregate-event-counts
-              version
-              query
-              [summarize_by query-options]
-              (:scf-read-db globals)
-              (:url-prefix globals))))}))
+    [""]
+    {:get (fn [{:keys [params globals]}]
+            (if (and (:datasource (:scf-read-db globals))
+                     (re-matches #"jdbc:hsqldb.*"
+                                 (.getJdbcUrl (:datasource
+                                                (:scf-read-db globals)))))
+              (http/json-response
+                {:error "The aggregate-event-counts endpoint does not support HSQLDB."}
+                http/status-not-found)
+              (let [{:strs [query summarize_by counts_filter count_by]} params
+                    distinct-options (validate-distinct-options! params)
+                    counts_filter' (json/parse-string counts_filter true)
+                    query-options (-> {:counts_filter counts_filter'
+                                       :count_by count_by}
+                                      (merge distinct-options))]
+                (log/warn
+                  (str "The aggregate-event-counts endpoint is experimental"
```

Comment:
if it's any indication we've had complaints about noisy logging in the past. I don't know that I agree we need a separate response -- my main hesitation about what we're currently doing is that it may look weird to PE users.